apps/src/lab2/views/components/Instructions/ResourcePanel/index.tsx

```diff
@@ -217,16 +210,12 @@ const ResourcePanel: React.FC<ResourcePanelProps> = ({
       );
     }
 
-    // The version history tab is hidden in read only mode with two exceptions:
-    // if the user is viewing an old version of the project, or if this is a teacher viewing
-    // a student's project (in which case they can view old versions, but not restore them).
-    // We never show the version history tab in widget view, as widget view is always read-only
-    // and therefore can never have version history.
-    // Note: We use the permanent read-only state captured at mount time to determine tab visibility.
+    // The version history tab is hidden in permanently read-only mode with the following exception:
+    // - if a teacher is viewing a student's project (in which case they can view old versions, but not restore them).
+    // Version history is also hidden on predict levels and widget view.
     const versionHistoryHidden =
-      (isPermanentlyReadOnlyRef.current &&
-        !isViewingOldVersion &&
-        !viewAsUserId) ||
+      (isPermanentlyReadOnly && !viewAsUserId) ||
+      isPredictLevel ||
```

Comment:
Even if teacher is viewing student work, never show version history on predict or widget levels.

Reply:
Nvm! I pushed a commit so that version history is now displayed on a predict level which is editable after response and response was submitted.